src/Chainweb/Chainweb/MinerResources.hs

```diff
@@ -64,11 +73,21 @@ withMiningCoordination logger enabled cutDb inner
     | not enabled = inner Nothing
     | otherwise = do
         t <- newTVarIO mempty
-        inner . Just $ MiningCoordination
+        fmap snd . concurrently (prune t) $ inner . Just $ MiningCoordination
             { _coordLogger = logger
             , _coordCutDb = cutDb
-            , _coordState = t
-            }
+            , _coordState = t }
+  where
+    prune :: TVar MiningState -> IO a
+    prune t = do
+        let !d = 600000000  -- 10 minutes
```

Comment:
Is 10 minutes long enough?

Reply:
use `runForever`?

Reply:
Or something similar

Reply:
Or is it fine if the miner crashes if this throws an exception?

Reply:
No that's bad. This is in the Mining Coordination logic, which can't crash.